components/src/core/user-menu/user-menu.tsx

```diff
@@ -0,0 +1,129 @@
+        avatar?: ViewStyle;
+    };
+};
+
+const useStyles = (
+    theme: ReactNativePaper.Theme
+): StyleSheet.NamedStyles<{
+    root: ViewStyle;
+    avatar: ViewStyle;
+}> =>
+    StyleSheet.create({
+        root: {
+            backgroundColor: theme.colors.surface,
+        },
+        avatar: {
+            width: 40,
+            height: 40,
+            borderRadius: 40,
+        },
+    });
+
+export const UserMenu: React.FC<UserMenuProps> = (props) => {
+    const theme = useTheme();
+    const {
+        avatar,
+        backgroundColor,
+        fontColor,
+        iconColor,
+        menu,
+        menuTitle,
+        menuSubtitle,
+        menuItems,
+        menuOpen = false,
+        toggleMenu = (): void => {},
+        styles = {},
+    } = props;
+    const [showBottomSheet, setShowBottomSheet] = useState(false);
+    const defaultStyles = useStyles(theme);
+
+    const openMenu = (): void => {
+        setShowBottomSheet(true);
+    };
+
+    const closeMenu = (): void => {
+        setShowBottomSheet(false);
+    };
+
+    const getMenu = useCallback((): JSX.Element => {
+        if (menu) {
+            return menu;
+        }
+
+        return (
+            <>
+                {menuTitle && (
+                    <>
+                        <InfoListItem
+                            title={menuTitle || ''}
+                            subtitle={menuSubtitle}
+                            customIcon={<View style={[defaultStyles.avatar, styles.avatar]}>{avatar}</View>}
+                            fontColor={fontColor}
+                            backgroundColor={backgroundColor}
+                        />
+                        <Divider />
+                    </>
+                )}
+                {menuItems &&
+                    menuItems.map((menuItem: InfoListItemProps, index: number) => (
+                        <InfoListItem
+                            key={index}
+                            IconClass={menuItem.IconClass}
+                            onPress={(): void => {
+                                closeMenu();
+                                if (menuItem.onPress) menuItem.onPress();
+                            }}
+                            iconColor={iconColor || menuItem.iconColor}
+                            fontColor={fontColor || menuItem.fontColor}
+                            backgroundColor={backgroundColor || menuItem.backgroundColor}
+                            {...menuItem}
+                        />
+                    ))}
+            </>
+        );
+    }, [menu, menuItems, menuTitle, menuSubtitle, iconColor, fontColor, backgroundColor]);
+
+    return (
+        <View style={[defaultStyles.root, styles.root]}>
+            <TouchableWithoutFeedback
+                onPress={menu ? (): void => toggleMenu() : (): void => openMenu()}
+                testID={'avatar'}
+            >
+                <View style={[defaultStyles.avatar, styles.avatar]}>{avatar}</View>
```

Comment:
Do we want to clone and modify the avatar like we do in react? To control things like the size?